react/features/recording/components/RecordingButton.web.js

```diff
@@ -0,0 +1,307 @@
+        };
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDialogClose = this._onDialogClose.bind(this);
+        this._onDialogToggle = this._onDialogToggle.bind(this);
+        this._onToggleRecordingChange
+            = this._onToggleRecordingChange.bind(this);
+        this._onToggleTranscriptionChange
+            = this._onToggleTranscriptionChange.bind(this);
+    }
+
+    /**
+     * Updates the toggled state depending on the _recordingButtonToggled
+     * prop.
+     *
+     * @param {Object} nextProps - The props that will be applied after the
+     * update.
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUpdate(nextProps) {
+        if (nextProps._recordingButtonToggled
+            !== this.props._recordingButtonToggled
+            && nextProps._recordingButtonToggled
+            !== this.state.isToggleRecordingChecked) {
+            this.setState({
+                isToggleRecordingChecked: nextProps._recordingButtonToggled
+            });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { _visible, tooltipPosition } = this.props;
+        const buttonConfiguration = {
+            ...DEFAULT_BUTTON_CONFIGURATION,
+            classNames: [
+                ...DEFAULT_BUTTON_CONFIGURATION.classNames,
+                this.state.showRecordingDialog ? 'toggled button-active' : ''
+            ]
+        };
+
+        const content = this._renderRecordingMenu();
+
+        return (
+            <AKInlineDialog
+                content = { content }
+                isOpen = { _visible && this.state.showRecordingDialog }
+                onClose = { this._onDialogClose }
+                position = { TOOLTIP_TO_DIALOG_POSITION[tooltipPosition] }>
+                <ToolbarButton
+                    button = { buttonConfiguration }
+                    onClick = { this._onDialogToggle }
+                    tooltipPosition = { tooltipPosition } />
+            </AKInlineDialog>
+        );
+    }
+
+    /**
+     * Hides the attached inline dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onDialogClose() {
+        this.setState({ showRecordingDialog: false });
+    }
+
+    /**
+     * Toggles the display of the dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onDialogToggle() {
+        this.setState({
+            showRecordingDialog: !this.state.showRecordingDialog
+        });
+    }
+
+    /**
+     * Updates the current known state of the toggle selection.
+     *
+     * @param {Object} event - The DOM event from changing the toggle selection.
+     * @private
+     * @returns {void}
+     */
+    _onToggleRecordingChange(event) {
+        this.setState({
+            isToggleRecordingChecked: event.target.checked
+        });
```

Comment:
One issue here is the assumption toggleRecording will always succeed. If recording fails or maybe is cancelled the button will remain toggled. The toggle should probably represent the source of truth in redux, assuming the update is snappy enough.